libyara.NET/Match.h

```diff
@@ -30,7 +30,7 @@ namespace libyaraNET {
 
             Data = gcnew array<uint8_t>(match->match_length);
             Marshal::Copy(
-                IntPtr(match->data),
+                IntPtr((void*) match->data),
```

Comment:
is this required? casting to `(void*)` feels like a workaround for what might be a deeper problem?

cc @kylereedmsft

Reply:
That's fine, IntPtr only 3 ctors, one takes a void*. maybe prefer `static_cast<void*>(match->data)` instead.

Reply:
Thanks Kyle. The static_cast is exactly what I want. Got some errors during compilation and added the void* case.

Reply:
Thanks Kyle. Looks like the older yara build returns a void* and this newer version returns const void* for the match data. Unfortunately, IntPtr does not take const void* as the ctor parameter.